elasticsearch/_async/helpers.py

```diff
@@ -439,24 +440,47 @@ async def async_reindex(
         :func:`~elasticsearch.helpers.async_scan`
     :arg bulk_kwargs: additional kwargs to be passed to
         :func:`~elasticsearch.helpers.async_bulk`
+    :arg op_type: Explicit operation type.
```

Comment:
Mention that the default is `index` but `create` is required for data streams and will auto-detect if `target_index` is a data stream.